### PR TITLE
fix(plus-17): L2 default child

### PR DESCRIPTION
**Pull request by iliapolo**
All L1 children are now called `Resource` to mark them as the default child of the L2.

BREAKING CHANGE: All L2 resource names will undergo a name change (e.g `test-chart-config-configmap-233db8e7` -> `test-chart-config-c3f7d3c0`)

Resolves https://github.com/awslabs/cdk8s/issues/373

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cdk8s-plus-17/src/config-map.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ export class ConfigMap extends Resource implements IConfigMap {
   public constructor(scope: Construct, id: string, props: ConfigMapProps = { }) {
     super(scope, id, { metadata: props.metadata });
 
-    this.apiObject = new k8s.KubeConfigMap(this, 'ConfigMap', {
+    this.apiObject = new k8s.KubeConfigMap(this, 'Resource', {
```
Comment (eladb) [inline packages/cdk8s-plus-17/src/config-map.ts:70]: We should add `Construct.DEFAULT_CHILD` or something like that, in the future.

File: packages/cdk8s-plus-17/test/config-map.test.ts [hunk 2]
```diff
@@ -1,6 +1,15 @@
 import { Testing } from 'cdk8s';
+import { Node } from 'constructs';
 import { ConfigMap } from '../src';
 
+test('defaultChild', () => {
+
+  const chart = Testing.chart();
+
+  expect(Node.of(new ConfigMap(chart, 'ConfigMap')).defaultChild).toBeTruthy();
```
Comment (eladb) [inline packages/cdk8s-plus-17/test/config-map.test.ts:9]: would it be better to assert `instanceof KubeConfigMap` or at least `ApiObject`.

BTW, should we rename `ApiObject` to `KubeObject`?

File: packages/cdk8s-plus-17/test/deployment.test.ts [hunk 3]
```diff
@@ -1,6 +1,15 @@
 import { Testing } from 'cdk8s';
+import { Node } from 'constructs';
 import * as kplus from '../src';
 
+test('defaultChild', () => {
```
Comment (eladb) [inline packages/cdk8s-plus-17/test/deployment.test.ts:5]: how do we enforce that such a test exists for all L2s? (for later)

File: packages/cdk8s/src/names.ts [hunk 4]
```diff
@@ -142,6 +122,26 @@ function omitDuplicates(value: string, index: number, components: string[]) {
   return value !== components[index-1];
 }
 
+function omitDefaultChild(value: string, _: number, __: string[]) {
+  return value.toLowerCase() !== 'resource' && value.toLowerCase() !== 'default';
```
Comment (eladb) [inline packages/cdk8s/src/names.ts:126]: I believe it is actually case sensitive in `constructs`.

File: packages/cdk8s/src/names.ts [hunk 5]
```diff
@@ -53,16 +53,7 @@ export class Names {
 
     components.push(calcHash(path, HASH_LEN));
```
Comment (eladb) [inline packages/cdk8s/src/names.ts:54]: We should use `Node.of(c).addr` instead of manually calculating the hash here.

**Review by eladb — CHANGES_REQUESTED** [review]

File: packages/cdk8s-plus-17/test/deployment.test.ts (hunk 3 above)
Comment (iliapolo) [reply packages/cdk8s-plus-17/test/deployment.test.ts:5]: Marked for later

**Review by eladb — APPROVED** [review]

File: packages/cdk8s/src/names.ts (hunk 5 above)
Comment (iliapolo) [reply packages/cdk8s/src/names.ts:54]: Noted

**Pull request merged**